sycl/include/CL/sycl/INTEL/esimd/esimd.hpp

```diff
@@ -480,6 +482,47 @@ template <typename Ty, int N> class simd {
     }
   }
 
+  /// @name Memory operations
+  /// TODO NOTE: These APIs do not support cache hint specification yet, as this
+  /// is WIP. Later addition of hints is not expected to break code using these
+  /// APIs.
+  ///
+  /// @{
+
+  /// Copy a contiguous block of data from memory into this simd object.
+  /// The amount of memory copied equals the total size of vector elements in
+  /// this object.
+  /// @param addr the memory address to copy from. Must be a pointer to the
+  /// global address space, otherwise behavior is undefined.
+  ESIMD_INLINE void copy_from(const Ty *const addr) SYCL_ESIMD_FUNCTION;
+
+  /// Copy a contiguous block of data from memory into this simd object.
+  /// The amount of memory copied equals the total size of vector elements in
+  /// this object.
+  /// Source memory location is represented via a global accessor and offset.
+  /// @param acc accessor to copy from.
+  /// @param offset offset to copy from.
+  template <typename AccessorT>
+  ESIMD_INLINE __ESIMD_ENABLE_IF_ACCESSOR(AccessorT, can_read, global_buffer,
+                                          void)
+      copy_from(AccessorT acc, uint32_t offset) SYCL_ESIMD_FUNCTION;
+
+  /// Copy all vector elements of this object into a contiguous block in memory.
+  /// @param addr the memory address to copy to. Must be a pointer to the
+  /// global address space, otherwise behavior is undefined.
+  ESIMD_INLINE void copy_to(Ty *addr) SYCL_ESIMD_FUNCTION;
+
+  /// Copy all vector elements of this object into a contiguous block in memory.
+  /// Destination memory location is represented via a global accessor and
+  /// offset.
+  /// @param acc accessor to copy from.
+  /// @param offset offset to copy from.
+  template <typename AccessorT>
+  ESIMD_INLINE __ESIMD_ENABLE_IF_ACCESSOR(AccessorT, can_write, global_buffer,
```

Comment:
NIT. Suggest aliasing the complex `enable_if` expression and use this alias here, like it's done with `EnableIfContiguous` in buffer.hpp:191.

Reply:
I think this one is more concise and readable, so I'd prefer this variant. Thanks for the NIT

Reply:
looks like I looked at the wrong place. Point of usage seems good indeed. will do